unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneBoundariesController/SceneBoundariesChecker.cs

```diff
@@ -2,21 +2,76 @@
 using DCL.Helpers;
 using DCL.Models;
 using UnityEngine;
+using System.Collections.Generic;
+using System.Collections;
 
 namespace DCL.Controllers
 {
     public class SceneBoundariesChecker
     {
-        protected ParcelScene scene;
+        [System.NonSerialized]
+        public float timeBetweenChecks = 1f;
 
-        public SceneBoundariesChecker(ParcelScene ownerScene)
+        HashSet<DecentralandEntity> entitiesToCheck = new HashSet<DecentralandEntity>();
+        HashSet<DecentralandEntity> checkedEntities = new HashSet<DecentralandEntity>();
+        Coroutine entitiesCheckRoutine = null;
+        float lastChecktTime;
+
+        public SceneBoundariesChecker()
+        {
+            entitiesCheckRoutine = SceneController.i.StartCoroutine(CheckEntities());
+            lastChecktTime = Time.realtimeSinceStartup;
+        }
+
+        // TODO: Improve MessagingControllersManager.i.timeBudgetCounter usage once we have the centralized budget controller for our immortal coroutines
+        IEnumerator CheckEntities()
+        {
+            while (true)
+            {
+                if (timeBetweenChecks > 0 && Time.realtimeSinceStartup - lastChecktTime < timeBetweenChecks) yield return null;
+
+                foreach (var entity in entitiesToCheck)
```

Comment:
It'd be nice to early test if `entitiesToCheck.Count > 0` before processing this foreach. Also, we should use the `using`/`MoveNext` construct as its a bit more memory efficient.